validateOffload.py

```diff
@@ -71,77 +70,81 @@ def run_ethtool_cmd(self, vf_rep: str) -> (int, int):
             if r.returncode != 0:
                 if "already exists" not in r.err:
                     logger.info(r)
-                    sys.exit(-1)
+                    raise RuntimeError(
+                        f"ValidateOffload error: {r.err} returncode: {r.returncode}"
+                    )
 
-        ethtool_output = r.out
-        rxpacket = self.parse_out_packet(ethtool_output, "rx_packet")
-        txpacket = self.parse_out_packet(ethtool_output, "tx_packet")
-        return (rxpacket, txpacket)
+        return r
+
+    def parse_packets(self, output: str, packet_type: str) -> int:
+        total_packets = 0
+        prefix = f"{packet_type}_queue_"
+        packet_suffix = "_xdp_packets:"
 
-    def parse_out_packet(self, output: str, prefix: str) -> Optional[int]:
         for line in output.splitlines():
             stripped_line = line.strip()
-            if stripped_line.startswith(prefix):
-                return int(stripped_line.split(":")[1])
-
-        return None
+            if prefix in stripped_line and packet_suffix in stripped_line:
+                packet_count = int(stripped_line.split(":")[1].strip())
+                total_packets += packet_count
+                
+        return total_packets
 
-    def run_st(self) -> RxTxData:
+    def run_st(self) -> Result:
         vf_rep = self.extract_vf_rep()
-        (rxpacket_start, txpacket_start) = self.run_ethtool_cmd(vf_rep)
-        sleep(self._duration)
-        (rxpacket_end, txpacket_end) = self.run_ethtool_cmd(vf_rep)
-
-        return RxTxData(
-            rx_start=rxpacket_start,
-            tx_start=txpacket_start,
-            rx_end=rxpacket_end,
-            tx_end=txpacket_end,
+        r1 = self.run_ethtool_cmd(vf_rep)
+        time.sleep(self._duration)
+        r2 = self.run_ethtool_cmd(vf_rep)
+
+        combined_out = f"{r1.out}--DELIMIT--{r2.out}"
+        combined_err = f"R1: {r1.err} R2: {r2.err}"
+        combined_returncode = max(r1.returncode, r2.returncode)
+
+        return Result(
+            out=combined_out, err=combined_err, returncode=combined_returncode
         )
 
-    def run(self, duration: int):
+    def run(self, duration: int) -> None:
         self.exec_thread = ReturnValueThread(target=self.run_st)
         self._duration = int(duration)
         self.exec_thread.start()
 
-    def stop(self):
-        logger.info(f"Stopping Get Vf Rep execution on {self.pod_name}")
-        r = self.exec_thread.join()
-
-        if self.iperf_pod_type == PodType.HOSTBACKED:
```

Comment:
We still need to handle this case, maybe need to overload stop for this plugin

Reply:
Lets handle this in generate output

Reply:
Do we need to handle it? The way the code currently handles it is with 
```python
    def extract_vf_rep(self) -> str:
        if self.iperf_pod_type == PodType.HOSTBACKED:
            logger.info(f"The VF representor is: ovn-k8s-mp0_0")
            return "ovn-k8s-mp0_0"
```